qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -13,17 +13,84 @@
 """Probability and phase functions for the mock IQ backend."""
 
 from abc import abstractmethod
-from typing import Dict, List, Any
+from typing import Any, Dict, List, Optional, Tuple
 import numpy as np
 from qiskit import QuantumCircuit
 from qiskit.exceptions import QiskitError
 from qiskit.providers.aer import AerSimulator
 from qiskit_experiments.framework import BaseExperiment
 
+# Define an IQ point typing class.
+IQPoint = Tuple[float, float]
+
 
 class MockIQExperimentHelper:
     """Abstract class for the MockIQ helper classes"""
 
+    def __init__(
+        self,
+        iq_cluster_centers: Optional[List[Tuple[IQPoint, IQPoint]]] = None,
+        iq_cluster_width: Optional[List[float]] = None,
```

Comment:
The defaults are defined in the `__init__` function instead of the signature. This is because there is an interdependence between `iq_cluster_centers` and `iq_cluster_width` in that their first dimensions must have the same length. If only `iq_cluster_centers` is provided, the widths are set as `[1.0, 1.0, ....]` with the correct length. The current defaults work and are used in existing testing code. For an example, see [test_readout_angle.py:36](https://github.com/Qiskit/qiskit-experiments/blob/82561acf86b407dcda0a9ec69fe18de2b0a592a2/test/test_readout_angle.py#L36) where `iq_cluster_centers` is set but `iq_cluster_width` is not. If we move the defaults to the signature, then both the centers and widths need to be provided every time a `MockIQExperimentHelper` instance is created for more than 1 qubit (or whatever default number of qubits we choose).

I have added a description of the default values to the `__init__` function of `MockIQExperimentHelper` (61e6d57). Let me know if the above approach is suitable or if the way defaults are handled by `MockIQExperimentHelper` should be changed.

Reply:
Ok, I'm fine with this as is.